openedx_learning/apps/authoring/collections/api.py

```diff
@@ -70,6 +87,90 @@ def update_collection(
     return collection
 
 
+def add_to_collections(
+    collections_qset: QuerySet[Collection],
+    contents_qset: QuerySet[PublishableEntity],
+) -> int:
+    """
+    Adds a QuerySet of PublishableEntities to a QuerySet of Collections.
+
+    Records are created in bulk, and so integrity errors are deliberately ignored: they indicate that the content(s)
+    have already been added to the collection(s).
+
+    Returns the number of entities added (including any that already exist).
+    """
+    collection_objects = []
+    object_ids = contents_qset.values_list("pk", flat=True)
+    collection_ids = collections_qset.values_list("pk", flat=True)
+
+    for collection_id in collection_ids:
+        for object_id in object_ids:
+            collection_objects.append(
+                CollectionObject(
+                    collection_id=collection_id,
+                    object_id=object_id,
+                )
+            )
+
+    created = []
+    if collection_objects:
+        created = CollectionObject.objects.bulk_create(
+            collection_objects,
+            ignore_conflicts=True,
+        )
+
+        # Update the modified date for all the provided Collections.
+        # (Ignoring conflicts means we don't know which ones were modified.)
+        collections_qset.update(modified=timezone.now())
+
+    return len(created)
```

Comment:
Oh, that's a fantastic callout. We should absolutely enforce it at the Learning Core layer, so people don't shoot themselves in the foot. It's very possible we'll eventually allow folks to add stuff from other LearningPackages into Collections, but we're going to have to be very deliberate when we do and think through the lifecycle issues when those LearningPackages disappear. We definitely don't want to let it happen by accident because other apps used Learning Core directly instead of going through the Libraries REST API.